### PR TITLE
Zephyr rtos port

### PR DESCRIPTION
**Pull request by PiotrZierhoffer**

### CODE REVIEW

File: src/civetweb.c [hunk 1]
```diff
@@ -162,6 +162,39 @@ mg_static_assert(sizeof(void *) >= sizeof(int), "data type size check");
...
+
+#include <libc_extensions.h>
+
+/* Max worker threads is the max of pthreads minus the main application thread
+ * and minus the main civetweb thread, thus -2
+ */
+#define MAX_WORKER_THREADS	(CONFIG_MAX_PTHREAD_COUNT - 2)
+
+#if defined(USE_STACK_SIZE) && (USE_STACK_SIZE > 1)
+#undef ZEPHYR_STACK_SIZE	USE_STACK_SIZE
```
Comment (PiotrZierhoffer) [inline src/civetweb.c]: undef?

File: src/civetweb.c [hunk 2]
```diff
@@ -18660,8 +18726,8 @@ mg_start(const struct mg_callbacks *callbacks,
 
 #if defined(ALTERNATIVE_QUEUE)
 	ctx->client_wait_events =
-	    (void **)mg_calloc_ctx(sizeof(ctx->client_wait_events[0]),
-	                           ctx->cfg_worker_threads,
+	    (void **)mg_calloc_ctx(ctx->cfg_worker_threads,
+				   sizeof(ctx->client_wait_events[0]),
```
Comment (PiotrZierhoffer) [inline src/civetweb.c]: some formatting changes happened here and below

Comment (tgorochowik) [reply src/civetweb.c]: Not really, this is actually a fix! :wink:

File: src/civetweb.c (hunk 1 above)
Comment (tgorochowik) [reply src/civetweb.c]: ops!

**Pull request closed without merging**